tests/baselines/reference/mappedTypeWithAsClauseAndLateBoundProperty2.js

```diff
@@ -4,3 +4,152 @@ export const thing = (null as any as { [K in keyof number[] as Exclude<K, "lengt
 
 //// [mappedTypeWithAsClauseAndLateBoundProperty2.js]
 export const thing = null;
+
+
+//// [mappedTypeWithAsClauseAndLateBoundProperty2.d.ts]
+export declare const thing: {

```

Comment:
this on the other hand is a direct result of the fact that we now can serialize those symbols properly, so the type for this could be emitted here